### ISSUE TITLE
Training Classification Model + tokenizer from Scratch

### ISSUE THREAD

**Issue by seyonechithrananda** [ThilinaRajapakse/simpletransformers#407]
**Describe the bug**
I've been looking through the repository and observed the `ClassificationModel` requires a pre-trained model as part of the args. Is there any chance we can take a model from scratch, train a tokenizer for our dataset and train the model on classification? Any help is really appreciated, I've really enjoyed using the simple transformers library for my research!

**Comment by ThilinaRajapakse** [comment]
Yes, you can do that. You can [train a language model](https://simpletransformers.ai/docs/lm-specifics/) and use the trained model for classification.

**Comment by leungi** [comment]
> Yes, you can do that. You can [train a language model](https://simpletransformers.ai/docs/lm-specifics/) and use the trained model for classification.

Philosophical question: in your experience, do you see a marked performance improvement (> 5% on relevant metric; e.g., F1 score) between these two fine-tuning options:

1. train a language model from scratch using domain-specific raw text, and condition it for classifying using doman-specific labeled text

2. use pre-trained language model, and condition it for classifying using doman-specific labeled text

**Comment by ThilinaRajapakse** [comment]
I can answer this empirically - option 2 is superior. That is why transfer learning is so powerful.

The philosophical question is whether it's worth the effort to fine-tune a pre-trained language model on the domain-specific text (i.e. fine-tune the language model itself) before training the model on the classification itself. In this case, I would suggest skipping the language model fine-tuning at first, and then coming back to it if the final results are not satisfactory.

**Comment by leungi** [comment]
Thanks for sharing your insights @ThilinaRajapakse 🙏

**Comment by leungi** [comment]
> **Describe the bug**
> I've been looking through the repository and observed the `ClassificationModel` requires a pre-trained model as part of the args. Is there any chance we can take a model from scratch, train a tokenizer for our dataset and train the model on classification? Any help is really appreciated, I've really enjoyed using the simple transformers library for my research!

@seyonechithrananda : you may use the `simpletransformers` or follow [this guide](https://huggingface.co/blog/how-to-train) by @huggingface to train the language model and plug the resulting model into `simpletransformers` `ClassificationModel` class for binary, multi-class and multi-label as per (really helpful) docs 🙂

`simpletransformers` is the easiest API I've found to date for the task you have at hand.

**Comment by seyonechithrananda** [comment]
Thanks @ThilinaRajapakse @leungi ! I've already trained a MLM model which is uploaded to HuggingFace's model hub. However, I've been getting this repeating issue of a training loss that doesn't converge at all with binary classification. My MLM model is trained on a domain-specific dataset, which is necessary as it deals with SMILES strings (a text-based molecular input). Its a RoBERTa-based MLM model. Do you have any suggestions as to potential reasons why the model doesn't converge at all?

See training loss plot from Weights and Biases below: 
[image](https://www.notion.so/ChemBERTa-ZINC-250k-sr-P53-unfiltered-07bad583c1b343bfbd881125610cd20c#05611c3517d14d8eb538ed2e6b258a5e)

**Comment by ThilinaRajapakse** [comment]
Judging by the discrepancy between accuracy and MCC score, is your data heavily unbalanced?

**Comment by seyonechithrananda** [comment]
> Judging by the discrepancy between accuracy and MCC score, is your data heavily unbalanced?
Correct! I've utilized the auto_weight = 'true' argument to try to combat that. The biggest thing I've noticed is in filtered datasets (where the distribution of negative-positive labels is less imbalanced) the model's training loss is a little more representative of a curve. Do you have any suggestions on how to deal with this issue?

**Comment by leungi** [comment]
@seyonechithrananda : in regular muti-variate regression, unbalanced data problem is typically handled by stratifying the train/test data set in equal proportions, with regards to the available classes. Another strategy I find helpful is under/oversampling the dominant and minor classes respectively.
